### ISSUE TITLE
Unhandled exception during scale down zookeeper instances

### ISSUE THREAD

**Issue by serrss** [strimzi/strimzi-kafka-operator#313]
There is an exception in TC during scaling down the number of instances of a zookeeper. An exception appears when the connection is lost for path: /brokers/topics.

### Step to reproduce:
1.Create "Apache Kafka (Persistent storage)"
1.1 Click "Add to project"
1.2 Click "Select from project"
2.3 Select "Apache Kafka (Persistent storage)" and click "Next"
1.4 Use default configurations:
_Number of Zookeper cluster nodes: 3
Number of Kafka cluster nodes: 3_
1.5 Click "Create" and wait for the message "Apache Kafka (Persistent storage) has been created"
1.6 Open page "Pods" and wait when all pods will have status "Running"
2. Open page "Config Maps" and open your cluster config
3. Edit numbers of zookeeper-nodes
3.1 Click "Actions"
3.2 Click "Edit YAML"
3.3 Set new value (1) for variable zookeeper-nodes and save
4. Check TC logs

### Actual result:
Log has an unhandled exception when the connection is lost

### Log:
```
[2018-03-13 11:04:13,968] INFO  <ClientCnxn  :876> [.local:2181)] Socket connection established to my-cluster-zookeeper.myproject.svc.cluster.local/172.30.118.74:2181, initiating session
--
  | [2018-03-13 11:04:13,973] INFO  <ClientCnxn  :1158> [.local:2181)] Unable to read additional data from server sessionid 0x3621f01bd320000, likely server has closed socket, closing socket connection and attempting reconnect
  | Mar 13, 2018 11:04:14 AM io.vertx.core.impl.ContextImpl
  | SEVERE: Unhandled exception
  | java.lang.RuntimeException: org.apache.zookeeper.KeeperException$ConnectionLossException: KeeperErrorCode = ConnectionLoss for for path: /brokers/topics
  | at io.strimzi.controller.topic.TopicsWatcher.lambda$start$2(TopicsWatcher.java:67)
  | at io.strimzi.controller.topic.zk.ZkImpl.lambda$null$9(ZkImpl.java:348)
  | at io.vertx.core.impl.ContextImpl.lambda$wrapTask$2(ContextImpl.java:344)
  | at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
  | at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:403)
  | at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:463)
  | at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:858)
  | at java.lang.Thread.run(Thread.java:748)
  | Caused by: org.apache.zookeeper.KeeperException$ConnectionLossException: KeeperErrorCode = ConnectionLoss for for path: /brokers/topics
  | at org.apache.zookeeper.KeeperException.create(KeeperException.java:99)
  | at org.apache.zookeeper.KeeperException.create(KeeperException.java:51)
  | at io.strimzi.controller.topic.zk.ZkImpl.mapResult(ZkImpl.java:135)
  | at io.strimzi.controller.topic.zk.ZkImpl.lambda$children$10(ZkImpl.java:343)
  | at org.apache.zookeeper.ClientCnxn$EventThread.processEvent(ClientCnxn.java:601)
  | at org.apache.zookeeper.ClientCnxn$EventThread.run(ClientCnxn.java:505)
```

**Comment by serrss** [comment]
Added all log file for this issue - [TC-log-file.txt](https://github.com/strimzi/strimzi/files/1806680/TC-log-file.txt)

**Comment by stanlyDoge** [comment]
Scaling down zookeeper cluster has the same issue as the scaling up. You have to be aware of rolling the leader as the last pod. @tombentley has described possible situations in some document but I have lost it :(

**Comment by serrss** [comment]
@stanlyDoge anyway, I couldn't reproduce this issue for latest version so I guess the issue was fixed in the scope of #1153
